service/utils.go

```diff
@@ -169,11 +200,24 @@ func versionResource(apiContext *api.ApiContext, catalogName string, template mo
 	links["template"] = URLEncoded(apiContext.UrlBuilder.ReferenceByIdLink("template", templateId))
 
 	upgradeVersionLinks := map[string]string{}
+	upgradeVersions := []model.Version{}
 	for _, upgradeVersion := range template.Versions {
 		if showUpgradeVersion(version, upgradeVersion, rancherVersion) {
 			route := generateVersionId(catalogName, template, upgradeVersion)
 			link := apiContext.UrlBuilder.ReferenceByIdLink("template", route)
 			upgradeVersionLinks[upgradeVersion.Version] = URLEncoded(link)
+			upgradeVersions = append(upgradeVersions, upgradeVersion)
+
+			if template.DefaultVersion == upgradeVersion.Version {
+				links["defaultUpgradeVersion"] = URLEncoded(link)
+			}
+		}
+	}
+
+	if _, ok := links["defaultUpgradeVersion"]; !ok {
+		defaultUpgradeVersionLink := defaultUpgradeVersionLink(upgradeVersions, catalogName, template, apiContext)
```

Comment:
Picks the latest valid upgrade version link if the default version is not a valid upgrade.